pyiron_atomistics/sphinx/output_parser.py

```diff
@@ -386,3 +386,89 @@ def get_convergence(self):
     def get_fermi(self):
         pattern = r"Fermi energy:\s+(\d+\.\d+)\s+eV"
         return np.array(re.findall(pattern, self.log_main)).astype(float)
+
+
+class SphinxWavesParser:
+    """Class to read SPHInX waves.sxb files (HDF5 format)
+
+    Initialize with waves.sxb filename, or use load ()
+
+    """
+
+    def __init__(self, fname=None):
+        if fname is not None:
+            self.load(fname)
+
+    def load(self, filename):
+        """Load a waves.sxb file (HDF5 format)
+
+        filename: file name
+        """
+        self.wfile = h5py.File(filename)
+        self._eps = None
+        self._read()
+
+    # Internal: check that wfile is set
+    def _check_loaded(self):
+        if not isinstance(self.wfile, h5py.File):
+            raise "No waves file loaded"
+
+    def _read(self):
+        self._check_loaded()
+        # load various dimensions
+        self.mesh = self.wfile["meshDim"][:]
+        self.Nx, self.Ny, self.Nz = self.mesh
+
+        self.n_states = self.wfile["nPerK"][0]
+        self.n_spin = self.wfile["nSpin"].shape[0]
+        self.k_weights = self.wfile["kWeights"][:]
+
+        # load the fft_idx to map from condensed psi to FFT mesh
+        # (different mapping per k)
+        self._fft_idx = []
+        self._n_gk = self.wfile["nGk"][:]
+        off = 0
+        for ngk in self._n_gk:
+            self._fft_idx.append(self.wfile["fftIdx"][off : off + ngk])
+            off += ngk
+
+        self.k_vec = self.wfile["kVec"][:]
```

Comment:
```suggestion
    @property
    def k_vec(self):
        return self.wfile["kVec"][:]
```

Essentially the same as above